client/control/components/RecipeForm.js

```diff
@@ -37,14 +42,51 @@ export class RecipeForm extends React.Component {
       action: pt.string.isRequired,
       arguments: pt.object.isRequired,
     }),
-  }
+    // route prop passed from router
+    route: pt.object,
```

Comment:
I didn't realize that `route` was coming from the router directly. I assumed that because this wasn't a connected component that it wasn't at the top level, but I guess I was wrong.